### DISCUSSION TITLE
Language: English or Portuguese? / Linguagem: Inglês ou Português?

### DISCUSSION THREAD

**Discussion by guilhermelowa** [okfn-brasil/querido-diario-comunidade#24] [General]
**ENGLISH VERSION** (texto em português abaixo):

# Should we use English or Portuguese as developing, documenting and discussing language?

During the project sprint last Saturday (15/08/20) the language issue was raised (again). The older contributors said that this issue is raised from time to time, but there isn't a documentation of this discussion on this project, so I decided to open this Issue with this intent - and I think a new discussion would be reasonable as well, as other Telegram members have suggested.

Even though there isn't a documented discussion in this repository, there is an [Issue on Serenata de Amor repository](https://github.com/okfn-brasil/serenata-de-amor/issues/87) which discuss this topic. I'll summarize it briefly (my take) and then add some arguments of my own.

## Pros English:
- The project is universal and English is the universal tech language
- Even though most participants are brazilians, the Telegram group is international (doesnt apply to Querido Diario yet, but might in the future)
- Serenata had international contributions from Day 0 - does this project have? I don't know
- International media coverage
- Transparency: If fairness or credibility issues are raised, any independent party can audit it - inclusing other countries
- Opportunity to learn / improve the English skills of participants

## Pros Portuguese:
- Project was born to solve a Brazil's issue
- Most contributors are brazilian (all so far for this project?)
- People have trouble with English, decreases communication overall

### Other relevant discussions:
- Discussing and documentation language could be different from developing language (as currently is on Telegram, but wasn't the case for Serenata)

In this project's case, we already discuss in Portuguese on Telegram, but the documentation and code are in English.

### My bits
I do appreciate the advantages of using English as default language, but I think the weight of not using Portuguese is too heavy to ignore. 

This isn't a issue only in tech, though. As [Vladimir Safatle puts](https://www.youtube.com/watch?v=9Uw9RHh7Km4&feature=youtu.be&t=4806), this production and evaluation in another language is reminiscent (I'd say comparable) to Catholic's Medieval Church, where knowledge and preaches were in Latin and couldn't be accessed by the people. We all criticize this now, but we are doing very similar - for very different reasons, of course. Safatle continues to say that roughly 3% to 4% of Brazil's population can read in English. "[The use of English] is a paralyzing pattern. Totally illusory. Instead of opening up the production to society, you close it with the promise of international circulation - a promise that wasn't fulfilled or hardly at all". Now, I also appreciate that Serenata's international circulation was indeed real and made possible by the language. And that's awesome! But does it justify the excluding of 96% of brazilian population?

Diversity was highly praised on the [sprint's feedback](https://docs.google.com/document/d/12HUod2fuQeO98PHi6p7O_w1FtuqDd4lbBl0PXnLGuNw/edit). People from different geographical locations, knowledge backgrounds, programming experience, age and so on, came together to work on the project. It was awesome. It is also praised on the OKBR website and job openings text. Again, awesome! But through developing, documenting and discussing (this last item is not the case here) in English, we are effectively excluding 96% of the population. It is a big loss on diversity, to say the least.

### My suggestion
For me this Issue was highly motivated by the topic (which I think is a very relevant discussion) and [Leticia's post on Telegram](https://labcodes.com.br/blog/pt/articles/por-que-internacionalizacao-e-localizacao-importam.html) about internationalizing the code.

I think refactoring the code might be too much. The post cited above does describe a strategy to do it incrementally, which is great. I think maybe adding some commentaries in Portuguese on the code could also work for code readability - without compromising the English. I don't have many thoughts on the code, but I think it makes sense to be in English, though. Very up for debate.

For the documentation, I really think it should be at least bilingual. I'm also willing to do this translation if the community deems relevant. Also some discussion involved here, but I think it would be better on another Issue.

### Extras
[Another relevant reading on this topic](https://github.com/corona-warn-app/cwa-backlog/issues/4). Please note that it is a German project, so the context is very different from ours. Thanks @anapaulagomes for the reading!

Both on Serenata's and on the above Issue, the discussion kinda heated up at some point. Let's try not to repeat the same mistake here? ;D

On that spirit, I should note that I don't like the word universal. I think global is much better. Who knows what aliens speak?

- - - - -

**VERSÃO EM PORTUGUÊS** (English version abo... you saw it.)
# Devemos usar Inglês ou Português como língua de desenvolvimento, documentação e discussão?

No sprint do projeto realizado no último sábado (15/08/20) a questão da linguagem foi levantada (novamente). Os contribuidores mais velhos disseram que essa questão é constantemente levantada, mas não há documentação sobre isso neste projeto, então eu decidi abrir essa Issue pra discutir isso e adicionar meu ponto de vista. Acredito que ainda seja uma discussão relevante, assim como outros sugeriram no sprint e no Telegram.

Ainda que não exista uma discussão documentada nesse repositório, há uma [Issue no Serenata de Amor](https://github.com/okfn-brasil/serenata-de-amor/issues/87) que discute o tópico. Vou tentar resumir (minha visão da discussão) e depois adicionar minha opinião.

## Pros do Inglês:
- O projeto é universal e inglês é a lingua universal na computação
- Apesar da maioria dos participantes serem brasileiros, o grupo é internacional (não se aplica ao Querido Diário ainda, mas talvez no futuro)
- O Serenata teve contribuições internacionais desdo primeiro dia. - O Querido Diário tem? Não sei
- Cobertura da imprensa internacional
- Transparência: se a justiça ou a credibilidade do projeto forem colocadas em questão, qualquer parte independente pode inspecionar o projeto - incluindo outros países

## Pros do Português:
- Projeto nasceu pra resolver um problema do Brasil
- A maioria dos contribuidores são brasileiros (todos neste projeto?)
- As pessoas têm dificuldade com Inglês, o que dificulta a comunicação

### Outras discussões relevantes
- A linguagem de discussão e documentação podem ser diferentes da do código em si - como é no Telegram hoje (discutimos em Português), mas não era o caso do Serenata

Apesar das discussões serem em Português no Telegram, o código e a documentação estão em Inglês aqui.

### Minha opinião
Eu entendo e aprecio as vantagens de usar Inglês como língua padrão, mas acredito que o peso de não usar Português é muito grande pra ser ignorado.

Isso não é uma questão só em tecnologia. Como [Vladimir Safatle coloca](https://www.youtube.com/watch?v=9Uw9RHh7Km4&feature=youtu.be&t=4806), a produção e avaliação em outra língua é remanescente (eu diria comparável) à prática da Igreja Católica Medieval de rezar a missa e escrever os livros em Latim, de forma inacessível ao povo. A gente critica essa atitude hoje, mas estamos fazendo bem parecido - por razões muito diferentes, claro. Safatle continua dizendo que aproximadamente 3% ou 4% da população brasileira consegue ler em Inglês. "[O uso de Inglês] é um padrão paralizante. Totalmente ilusório. Ao invés de abrir a produção à sociedade, vocÊ fecha com uma promessa de circulação internacional - uma promessa que não é cumprida ou muito pouco cumprida". Eu entendo que a circulação internacional do Serenata foi real e tornada possível pelo Inglês. E isso é muito massa! Mas isso justifica a exclusão de 96% da população brasileira?

A diversidade foi muito elogiada no [feedback do sprint](https://docs.google.com/document/d/12HUod2fuQeO98PHi6p7O_w1FtuqDd4lbBl0PXnLGuNw/edit). Pessoas de diferentes regiões geográficas, áreas do conhecimento, experiências em programção, idade e por aí vai, se reuniram pra trabalhar no projeto. Foi muito massa. A diversidade também é muito enaltecida na OKBR e no texto da recente vaga de emprego. De novo, muito massa! No entanto, desenvolvendo, documentando e discutindo em Inglês (esse último não é o caso pra esse projeto) estamos efetivamente excluindo 96% da população brasileira. É uma grande perda em diversidade, pra dizer o mínimo.

### Minha sugestão
Pra mim essa Issue foi muito motivada pelo assunto (que pra mim é uma discussão muito relevante) e pelo [post de Leticia no Telegram](https://labcodes.com.br/blog/pt/articles/por-que-internacionalizacao-e-localizacao-importam.html) sobre internacionalização do código.

Acho que refatorar o código pode ser demais. O post acima cita uma estratégia pra ir mudando aos poucos, que parece boa. Talvez adicionar comentários em Português resolva a legibilidade do código - sem atrapalhar o Inglês. Não tenho muita opinião sobre o código, mas acho que faz sentido continuar em Inglês. Aberto a muito debate eu diria.

Pra documentação, eu realmente acho que deveria ser ao menos bilingue. Tô disposto a fazer a tradução se a comunidade achar relevante. Também há algumas discussões pra realizar isso, mas acho que é melhor em outra Issue.

### Extras
[Outra Issue relevante sobre esse tópico](https://github.com/corona-warn-app/cwa-backlog/issues/4) (em Inglês). É um projeto na Alemanha, então o contexto é diferente. Obrigado à @anapaulagomes 

Tanto no do Serenata como nessa Issue acima, a discussão esquentou em um momento... Vamos tentar não repetir o mesmo erro! ;D

No mesmo espírito, queria dizer que não gosto da palavra universal. Acho que global é muito melhor. Quem é que sabe o que é que os ETs falam?

**Comment by ogecece** [comment]
## English version

First, congratulations on the extensive post. As a long-post-writer myself, I appreciate when I see one in the wild :) Very good points were made and thanks for citing some sources that can broaden our discussion. Also, thanks for the effort in translating it to Portuguese.

I'd like to open with my conclusions first and then try to explain why:

_Documentation should be bilingual. Code should be in English. Discussions? Not sure._

### Documentation should be bilingual

To summarize, I agree with what you said.

Just adding something: [this review](http://igor.pro.br/publica/papers/IST_SysReview_PrePrint.pdf) talks about four main points on documentation as problems of newcomers contributing to open source. Although I couldn't find something specific about translation-related issues, I think this can give a little more strength to the argument that documentation is important. And we may be scaring off people by not having at least documentation in Portuguese. To top it off, this project has a lot of casual contributors and I don't know if the impact is better or worse because of it.

A workflow that comes to my mind would be: before merging every PR that changes documentation an issue in the other languages that were not changed should be created as well (thus, allowing for people that don't know English to make changes and then English speakers could step in and document it). I don't know if that has been tested before and if anyone knows a better workflow I'd like to know about it too but I don't think this should be too difficult.

But as you said, discussions on how to do it could be made in another issue.

### Code should be in English

I don't know how code comments could be done in a way that doesn't get in the way of readability. I propose we have a set of documents that translates the terms used (gazette, legislative, etc.) and let the code remain like it is right now.

### Discussions? Not sure

To start it off, I don't think Portuguese-only speakers should be confined to Telegram. It doesn't provide enough resources to track discussions and contribute in the same way the Github repo does. Discussions track stuff that happened in a way that commit messages and static documentation can't do.

Not to say that Telegram shouldn't be used. Discussions in Telegram introduce a more casual discussion channel and in general those are more active than the repo. That's fine actually, it lowers the entrance barrier and can engage newcomers. The main difference between both communication channels should be only that.

Ideally, we would have every post made like you did it, but I think that's not achievable. Confining Portuguese-only speakers to one communication channel is not good, but it's better than not having them at all. Having some posts in English and some Portuguese is confusing and does not solve the issue. I think this topic engages more with the trade-off "internationalization vs localized engagement" than the others. I don't know, this is hard :)

## Versão em Português... ~Versão brasileira. Herbert Richers~

Primeiramente, parabéns pelo post extenso. Como escritor de posts longos eu acho legal encontrar outro por aí :) Bons argumentos foram feitos e obrigado por citar fontes que podem expandir nossa conversa. E obrigado pelo trabalho de traduzir o post para Português.

Eu gostaria de introduzir minhas conclusões primeiro e depois tentar explicar o porquê delas:

_Documentação deveria ser bilíngue. Código deveria ser em Inglês. Discussões? Não tenho certeza._

### Documentação deveria ser bilíngue

Para resumir, eu concordo com o que você disse.

Só adicionando uma coisa: [esta review](http://igor.pro.br/publica/papers/IST_SysReview_PrePrint.pdf) menciona quatro pontos principais sobre documentação como problemas enfrentados por iniciantes em projetos open source. Embora eu não tenho conseguido algo específico sobre problemas relacionados a tradução, eu acho que isso pode dar um pouco mais de força ao argumento de que documentação é importante. E, nós podemos estar afugentando pessoas por não ter pelo menos uma documentação em português. Além disso, esse projeto tem muitos contribuidores casuais e eu não sei se o impacto sentido por conta disso seria maior ou menor.

Um fluxo de trabalho que imagino seria: antes de mesclar cada PR que altera a documentação, uma issue nas outras línguas que não foram alteradas deveria ser criadas também (assim permitindo que pessoas que não sabem inglês possam fazer alterações e então pessoas que falam inglês poderiam documentá-las). Eu não sei se isso já foi testado e se alguém conhece um fluxo melhor eu também gostaria de conhecê-lo mas eu não acho que deveria ser algo muito difícil.

Mas, como você disse, discussões sobre como fazer isso poderiam ser feitas em outra issue.

### Código deveria ser em Inglês

Eu não sei como comentários poderiam ser feitos de uma maneira que não atrapalhe a legibilidade. Proponho que tenhamos um conjunto de documentos que traduzam os termos utilizados (gazette, legislative, etc.) e deixe o código como está.

### Discussões? Não tenho certeza

Pra começcar, eu acho que não é legal pessoas que falam apenas português ficarem limitadas ao uso do Telegram. Ele não provê recursos suficientes para rastrear discussões e contribuir da mesma maneira que o repo Github faz. Discussões são rastreio de coisas que aconteceram de uma maneira que mensagens de commit e documentação estática não são.

Não quero dizer que o Telegram não deveria ser utilizado. Discussões no Telegram são um meio de comunicação mais casual e em geral tem mais atividade do que o repo. Isso é bom na verdade, pois diminui a barreira de entrada e pode engajar novatos. A principal diferença entre ambos os meios de comunicação deveria ser só essa na real.

Idealmente, teríamos todo post feito como o seu, mas não acho que isso seja factível. Confinar pessoas que falam apenas português a apenas um meio de comunicação não é legal, mas é melhor do que não tê-las. Ter alguns posts em inglês e alguns em português é confuso e não resolve o problema. Eu acho que este tópico se relaciona mais com a relação "internacionalização vs engajamento localizado" do que os outros. Eu não sei, isso é cabuloso :)

**Comment by guilhermelowa** [comment]
# ENGLISH

Thanks for compliments and for your reply! (: Not sure if I like writing long posts (probably do), but I like being thorough hahaha

The review you brought is interesting indeed. Thanks!

About the internationalization vs localized engagement, this is indeed curious. I wonder what the proportion is for long-term internationalized projects, like Serenata.

So should I wait for more people to agree with us or should I just start a PR for the translation of the documentation?

# Português

Obrigado pelo elogio e pela resposta! (: Não sei se eu gosto de escrever posts grandes (acho que sim), mas eu gosto de ser completo uaheofuaef (traduzi a risada sem querer)

A review que você trouxe é muito massa mesmo! Obrigado!

Sobre a internacionalização vs engajamento local, tb acho isso curioso. Qual será que é a proporção dessa contribuição pra projetos já estabelecidos e internacionacionalizados, tipo o Serenata?

Você acha que eu devo esperar mais pessoas concordarem ou eu abro um PR pra traduzir a documentação?

**Comment by ogecece** [comment]
## English version

Thanks for your comments :)

> Not sure if I like writing long posts (probably do), but I like being thorough hahaha

ONE OF US! ONE OF US!

> About the internationalization vs localized engagement, this is indeed curious. I wonder what the proportion is for long-term internationalized projects, like Serenata.

Very good research topic. I would like to read about it.

> So should I wait for more people to agree with us or should I just start a PR for the translation of the documentation?

I mean. Maybe it could be blocked until the agreement is not yet reached. Also, even if it is reached, a workflow of translation is not defined yet. But a PR is surely more concrete and could advance the conversation. Go with what you feel to be right :)

## Versão em Português

Obrigado pelos comentários :)

> Não sei se eu gosto de escrever posts grandes (acho que sim), mas eu gosto de ser completo uaheofuaef (traduzi a risada sem querer)

UM DE NÓS! UM DE NÓS!

(morri com essa tradução da risada)

> Sobre a internacionalização vs engajamento local, tb acho isso curioso. Qual será que é a proporção dessa contribuição pra projetos já estabelecidos e internacionacionalizados, tipo o Serenata?

Um tema de pesquisa muito bom. Gostaria de ler mais sobre.

> Você acha que eu devo esperar mais pessoas concordarem ou eu abro um PR pra traduzir a documentação?

Então. Talvez ele possar ficar bloqueado até que haja concordância. E mesmo se for atingido, um fluxo de trabalho de tradução ainda não for definido. Mas um PR certamente é mais sólido e pode avançar a discussão. Faz o que seu coração manda :)

**Comment by jvanz** [comment]
#### English version

Let me share my thouhgs with you....

Yes, I agree that the choice of using English as the default language make harder to the vast majority of the population help the project. I do not like that either. But I think we should have more concrete data about the topic to take a final decision. I believe we need to know more about some things :

- What kind of material the contributor is looking for in Portuguese? Documentation about the Querido Diário project as a whole? Code comments? Code documentation?
- Is possible to contribute in code without knowing at leas a little bit of English? In other words, when the contributor look for additional info about the tools that we use, will they found documentation in Portuguese? If they cannot, will we support them on that as well?
- Do we have international interesting in the project? I mean, how the spiders, the only code present in the repo right now, can help people overseas?

On top of my head, I think we should continue keep the code stuff and comment in English and have bilingual documentation.We should not reject issues/code contributions just because the author send everything in Portuguese. This will create more overhead over the maintenance/review side. Because the reviewer/maintainer should be responsible for keep everything in both languages and help in the translation. However, I believe it worth the effort. 

#### Versão em Português.

Deixe-me compartilhar meus pensamentos com você ....

Sim, concordo que a escolha de usar o inglês como idioma padrão torna mais difícil para a grande maioria da população ajudar o projeto. Eu também não gosto disso. Mas acho que devemos ter dados mais concretos sobre o assunto para tomar uma decisão final. Acredito que precisamos saber mais sobre algumas coisas:

- Que tipo de material o colaborador procura em português? Documentação sobre o projeto Querido Diário como um todo? Comentários de código? Documentação do código?
- É possível contribuir em código sem saber pelo menos um pouco de inglês? Em outras palavras, quando o colaborador procurar informações adicionais sobre as ferramentas que usamos, ele encontrará a documentação em português? Se eles não puderem, iremos apoiá-los nisso também?
- Temos interesse internacional no projeto? Quero dizer, como os spiders, o único código presente no repo no momento, podem ajudar as pessoas no exterior?

Em cima da minha cabeça, acho que devemos continuar mantendo o código e os comentários em inglês e ter a documentação bilíngue. Não devemos rejeitar questões / contribuições de código só porque o autor manda tudo em português. Isso criará mais sobrecarga no lado da manutenção / revisão. Porque o revisor / mantenedor deve ser responsável por manter tudo nos dois idiomas e ajudar na tradução. Porém, acredito que vale a pena o esforço.

@sergiomario acho que a OKBR como uma organização ativa no projeto deveria compartilhar o que pensa. Ainda mais porque vocês sejam os mais interessados em algum compartilhamento internacional para captação de recurso ou qualquer outra coisa.

**Comment by gabriel19913** [comment]
@jvanz Achei válido seus comentários, apesar de não ser da organização, acho que faz sentido ter documentação bilíngue e os códigos/comentários em inglês. Já existe algo no sentido de traduzir a documentação? Ou vocês ainda estão discutindo se esse seria mesmo o caminho a ser seguido? Pois gostaria de contribuir com a tradução.

**Comment by sergiomario** [comment]
#### English version

Thanks @guilhermelowa for raising this issue in such a clear and detailed way. You were very generous with the project in raising this discussion. I would also like to take this opportunity to thank everyone who put their views on the thread.

The Querido Diário project is very large and ambitious, in several spheres, the trend is to have many repositories with APIs, data scraping, automation, data analysis and others. In addition, we have a legacy of activities carried out and documented in English for just over 3 years. In view of what has been mentioned, I think that **the path will be to create documentation outside of github and bilingual**. There are many topics of public interest that need to be detailed on a platform with a specific focus on project documentation.

**I see a subtle difference between community communication channels like Telegram's mailing lists and groups with respect to Github**. Unlike Operation Serenata de Amor, the Querido Diário project maintains its communication on these platforms in Portuguese, as we believe it is more efficient. Here at github, communication is even more asynchronous than net conversations from a message group, I think people can use translation tools or have more time to reason in another language.

**The issue of international contribution goes beyond receiving contributions from people from other countries**, but also preparing Brazilians to contribute to other large and important international projects. I also see OKBR's projects as a channel for training people.

**Impact on international media and partnerships**: Not only the project documentation, but the Querido Diário website must have its content all in English and Spanish. Latin America contains the closest society and they must have access to our advances in social control. ** Querido Diário's first funding came from an organization in Latin America **, [the Empatía program](https://www.ok.org.br/noticia/querido-diario-selecionado-por-programa-de-promotion-of-use-of-ia-for-development-in-latin-america/).

I vote to keep the documentation, commits and code of this repository in English, as a change now would generate a strong dedication of energy in changing culture and communication. In addition, we would have a whole history in English which could cause even more confusion for future contributors.

It is important to have these discussions in new Brazilian social control projects. The Operação Serenata de Amor and the Querido Diario should serve as examples for us.

#### Versão em Português

Primeiramente, muito obrigado @guilhermelowa por levantar essa questão de forma tão clara e detalhada. Você foi muito generoso com o projeto levantando essa discussão. Aproveito também para levar o agradecimento a todas as pessoas que colocaram seus pontos de vista na thread.

O projeto Querido Diário é muito grande e muito ambicioso, em várias esféras, de hoje em diante a tendência é termos N repositórios com APIs, raspagem de dados, automação, análise de dados e outros. Além disso, temos um legado de atividades executadas e documentadas em inglês de pouco mais de 3 anos. Em virtude do que foi mencionado, adianto que **o caminho será criarmos uma documentação fora do github e bilíngue**. Pois são muitos temas de interesse público que necessitam de detalhamento em uma plataforma com o foco específico para documentação projetos. 

**Eu vejo uma sutil diferença entre os canais de comunicação da comunidade como as listas de emails e grupos do Telegram com relação ao Github**. Diferente da Operação Serenata de Amor, o projeto Querido Diário mantém sua comunicação nessas plataformas em português, pois acreditamos ser mais eficiente. Aqui no github é uma comunicação ainda mais assíncrona do que conversas líquidas de um grupo de mensagem, penso que as pessoas podem se apoiar em ferramentas de tradução ou ter mais tempo para raciocinar em outra língua.

**A questão de contribuição internacional vai além de recebermos contribuições de pessoas de outros países**, mas também preparar brasileiras para contribuirem para outros grandes e importantes projetos internacionais. Eu vejo os projetos da OKBR também como um canal de formação de pessoas. 

**Impacto na mídia internacional e parceirias**: Não só a documentação do projeto, mas o site do Querido Diário deve ter seu conteúdo todo em inglês e espanhol. A América Latina contém a sociedade mais próxima e devem ter acesso aos nossos avanços em controle social. **O primeiro financiamento do Querido Diário veio de uma organização da América Latina**, [o programa Empatía](https://www.ok.org.br/noticia/querido-diario-selecionado-por-programa-de-promocao-do-uso-de-ia-para-o-desenvolvimento-na-america-latina/).

Eu voto por manter em inglês a documentação, commits e código desse repositório, pois uma mudança agora iria gerar uma forte dedicação de energia na mudança de cultura e comunicação. Além disso, teríamos todo um histórico em inglês podendo gerar mais confusão ainda para futuros navegantes.

Eu acredito que temos de ter essas discussões em novos projetos de controle social brasileiros. Que o Serenata e o Querido Diário sirvam de exemplos para nós.

**Comment by jvanz** [comment]
> @jvanz Achei válido seus comentários, apesar de não ser da organização, acho que faz sentido ter documentação bilíngue e os códigos/comentários em inglês. Já existe algo no sentido de traduzir a documentação? Ou vocês ainda estão discutindo se esse seria mesmo o caminho a ser seguido? Pois gostaria de contribuir com a tradução.

Acho que o último comentário do @sergiomario já responde sua pergunta. Certo? (desculpe pela demora)

**Comment by alexandrevicenzi** [comment]
My 2 cents if it counts for anything.

Developers are supposed to know English, Python uses it, all docs are in English, a lot of good books are also, even tho some books have translation not many are great on that.

Code and documentation in the code, like comments, should be in English in my view.

Documentation intended for people that are not contributing, AKA users, should be at very least in English and Portuguese, we want people to know what this project is about and also how to use it.

FAQ, Wiki, user documentation, public website should be in both languages, no need to be in German or Spanish at this moment, the aim is Brazil, in the future it can possibly grow and we may rethink our decision on the choices of languages.

I'm too lazy to translate everything, so...

Código e documentação relacionada deve ser em Inglês.
FAQ, Wiki, documentação para usuário final e futuro site com os diários deve ser em Inglês e Português.

**Comment by alexandrevicenzi** [comment]
Going a bit on @sergiomario comments, it does make some sense to have Spanish as well, it's a valid point, Brazil is the only country to use PT in South America, the project could gain some media traction from other countries if in Spanish.

On the communications channels that I forgot before, I would suggest keeping Telegram in pt-br, or creating two, or mixing languages, but I think it would be ok for now to say we will use pt-br to be more inclusive.

I'm not sure how practical it can be, but I find Indigenous languages something interesting, remember that there is a small minority that speaks some of those languages, I find it interesting to show that we value these communities, but as I said, not sure how practical it would be to translate.

**Comment by guilhermelowa** [comment]
**English**

Thanks for the reply, @sergiomario (: 

Looking forward to that documentation outside GitHub. Willing to help translating it when its time

**Português**

Obrigado pela resposta, sergiomario! (:

Fico no aguardo então pela documentação externa ao GitHub e estou disposto a ajudar na tradução quando chegar a hora

**Comment by guilhermelowa** [comment]
> Developers are supposed to know English, Python uses it, all docs are in English, a lot of good books are also, even tho some books have translation not many are great on that.

I think it would be great if all developers knew English, @alexandrevicenzi! Unfortunately, that's not the case. I think the main point of my argument was precisely that. Only about 5% of Brazil's population knows English. I don't have figures for devs (and I think numbers are much better among us), but given the extreme low percentage in the general population I take that English is still very exclusive - maybe not so much for the code, given its constraints.

As you said, most good resources are on English. But still, we have lots of resources on Portuguese (and translations, as you mentioned). Why is that? It would be much better for the writer to write the resource in English: it would drastically increase the potential market. My guess is that this happens because: 1- the author doesn't feel comfortable writing in English; 2- there's a market demand for content in Portuguese. If I'm right, both indicate that language is still a big barrier to overcome for our devs. 

This certainly has confirmation in my personal experience. Most my colleagues struggle with English. My guess is that this is the normal scenario outside Southeast top universities, where internationalization is much bigger and personal income too. Here I may be falling into the very dangerous trap to extrapolate my reality to generalize. But if my guesses are right, then it would be even more dangerous for someone in a top University to extrapolate as well.

**Comment by camilafracaro** [comment]
I wrote my documentation (#301) not knowing about the subject of this issue, but I was on the sprint and remember the subject. I did on mine to be more inclusive. 
I do understand that English should be the general language of the code, but I also agree with @guilhermelowa that a lot of Brazilians don't know English that well and we get even "shyer" when we're writing it, because people might judge you and it's kind of there forever. I think at least documentation should be bilingual to help people start helping the project and I don't think that there's so much documentation and this project is mainly for the Brazilian people. 

Some other points: 
- We're trying to include people that don't know (much), but the PRs would have to be in English anyway. Should the ones that speak English help translate when replying their PRs? This is an extra toll on who's accepting PRs.
- Every time that you change the documentation, you'll need to remember to change on the Portuguese version too.

**Comment by alexandrevicenzi** [comment]
Precisa decidir qual é o foco do projeto. É Brasil? Fala Português então.

Mas alguém pode querer usar um dia para outra coisa, podemos expandir e tal. Ok, mas e até la? Sem contar que os spiders não são reusáveis, talvez algumas partes não fazem sentido algum ser em Inglês.

Se o foco é cobrir mais de 5000 cidades vai precisar de muita gente motivada para escrever código, mesmo que a pessoa saiba Inglês, penso que pelo fato de ser Brasileiro não vai reclamar de o projeto estar em pt-br.

Pensando nesses pontos eu diria que o nosso idioma é o que faz sentido, no futuro pode-se traduzir para Inglês. Já não tem recursos de sobra, ficar gastando tempo em tradução é complicado.

**Obs:** Escrevi em Português pra enfatizar que faz mais sentido, se pensando no Brasil e sua população.

**Comment by alexandrevicenzi** [comment]
We should all agree on what makes more sense if we look on those who contribute, where the focus is, and for whom this project is meant to.

I don't want to throw more gasoline in the fire pit, but it seems that we are going on a long road that will never reach a destination.

I do think that English and Spanish make sense, but given the constraints and all comments regarding the lack on English knowledge, it seems to make more sense if we stick with Portuguese for a while.

**Comment by rodbv** [comment]
Pessoal, eu sou do time "código em inglês, o resto em português", inclusive issues e PRs. Nem vou me alongar pq acho que os argumentos do Mario e Giulio já cobriram muito bem. Uma questão de coerência até com a finalidade do projeto, acessibilidade. 

Queria saber se posso começar a fazer PRs e issues em português.

**Comment by ogecece** [comment]
## Comentário em Português

Olar pessoas! Pra dar mais uma movimentada nesse tópico, procurei alguém que tivesse pesquisa desenvolvida na área de contribuições open source pra bater um papo e saber a opinião sobre o caso desse projeto. Assim, consegui uma conversa com o [Igor Steinmacher](http://igor.pro.br/) no começo desse mês.

[A ata com tópicos anotados, vídeo e áudio está aqui (em português).](https://www.notion.so/Ata-do-papo-sobre-documenta-o-com-Igor-Steinmacher-afe5420b68e7434ca706e0c5757eafcb)

O papo foi bem legal, vários tópicos já mencionados aqui foram abordados e já deu bastante pano pra manga. Estamos em busca agora de compilar os dados e pesquisas que sustentam o que foi falado lá. É isso :) vamos seguindo!

## English version

Hi peoples! To shake things up on this topic, I looked for someone who had done research on open source contributions to have a talk and know their opinion about this project's case. That's when I got to talk to [Igor Steinmacher](http://igor.pro.br/) at the beginning of this month.

[The meeting notes with annotated topics, video and audio is here (in Portuguese).](https://www.notion.so/Ata-do-papo-sobre-documenta-o-com-Igor-Steinmacher-afe5420b68e7434ca706e0c5757eafcb)

The conversation was really nice, many aforementioned topics were talked about and it was very insightful. Now we are looking into compiling the data and research that supports those arguments. That's it :) Let's keep going!
